lib/azure/armrest/storage_account_service.rb

```diff
@@ -95,41 +97,27 @@ def list_all
       #   sas = Azure::Armrest::StorageAccountService(config)
       #
       #   sas.create(
-      #     :name     => "yourstorageaccount1",
-      #     :location => "West US",
-      #     :type     => "Standard_ZRS",
-      #     :tags     => {:YourCompany => true}
+      #     "yourstorageaccount1",
+      #     :location   => "West US",
+      #     :tags       => {:YourCompany => true}
+      #     :properties => {:accountType => "Standard_ZRS"},
       #   )
       #
       # For convenience you may also specify the :resource_group as an option.
       #
-      def create(options = {}, rgroup = armrest_configuration.resource_group)
-        rgroup ||= options[:resource_group]
+      def create(name, options = {}, rgroup = armrest_configuration.resource_group)
+        rgroup = options.delete(:resource_group) || rgroup
```

Comment:
`options` is directly passed to api call, we don't need to provide such connivence to set `resource_group` here. It can always be passed in as the `rgroup` option.